### PR TITLE
Doc: additional clarity #58 #59

### PR DESCRIPTION
**Pull request by katriendg**
* Updated readme doc to clarify pre-requisites and post-deployment validation, updated diagram
* Updated developer doc on using devcontainer and some small changes

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -19,7 +19,11 @@
 	// "forwardPorts": [],
 
 	// Configure tool-specific properties.
-	// "customizations": {},
+	"customizations": {
+		"vscode": {
+			"extensions": ["ms-azuretools.vscode-bicep", ]
+		}
```
Comment (suneetnangia) [inline .devcontainer/devcontainer.json:25]: Should we have other standard extensions here as well? e.g. Markdown validation, spell checkers etc.

**Review by suneetnangia — COMMENTED** [review]
Added some minor comments for your review and amendments. Thanks.

File: README.md [hunk 2]
```diff
@@ -88,16 +88,56 @@ Solution components are split into three layers from deployment perspective:
...
+        - Get the logs for the Data Gateway module: `kubectl logs -l app=data-gateway-module -n edge-app1`. You should see some logs about messages published.
+    - You can also further view other workloads by reviewing deployments in the namespaces `edge-core`, `edge-infra` and `azure-arc`.
+    - Connecto the L2 cluster and review the application (and their logs) deployed to `edge-app1` namespace.
+
+### Clean-up Resources
+
+> **Warning**
+> Please ensure you delete the deployed Azure resources after testing. Azure deployed resources will be charged into your subscription and can run up due to multiple AKS clusters being provisioned.
+
+`Invoke-WebRequest -Uri "https://raw.githubusercontent.com/azure-samples/distributed-az-edge-framework/main/deployment/remove-dev-resources.ps1" -OutFile "./remove-dev-resources.ps1" && ./remove-dev-resources.ps1 -ApplicationName <ApplicationName>`
```
[the remark this answers is not in the document]
Comment (katriendg) [reply README.md:137]: This works well for non dev deployments, I'm using it always when cleaning up my cloud shell test deployments.
However, what do you think about renaming this file in `cleanup-az-resources.ps1` and updating the references to it in both docs?

File: README.md [hunk 3]
```diff
@@ -88,16 +88,56 @@ Solution components are split into three layers from deployment perspective:
...
+- Three resource groups with each having their own AKS cluster, separate and peered virtual networks and Azure Arc connected.
+- One resource group with Event Hub resource. Messages from the edge workload simulators will be published to this Event Hub.
+- All resources will be provisioned in West Europe region. 
+
+Within the sample AKS clusters, several workloads are deployed as depicted in the [Nested Topology](#nested-topology) section above. Application workloads are deployed using Arc's Flux extension (GitOps).
+The lower networks are denied outbound internet access and have to route traffic through to the upper layers until it reaches public internet on Level 4.
+
+To validate the deployment of the default resources and sample workloads you can validate a few things:
+
+- In Azure:
```
[the remark this answers is not in the document]
Comment (katriendg) [reply README.md:121]: It's a valid point. 
My proposal is add a very simple validation step with EH and then move this section with a bit more details to the developer setup doc. So not a separate doc but a validation more detailed step in developer setup. Wdyt?

File: deployment/deploy-dev.md
[the remark this answers is not in the document]
Comment (katriendg) [reply deployment/deploy-dev.md:1]: Done

File: README.md
[the remark this answers is not in the document]
Comment (katriendg) [reply README.md:1]: Done

File: README.md [hunk 4]
```diff
@@ -88,16 +88,56 @@ Solution components are split into three layers from deployment perspective:
...
 Run the following command in the [Azure Cloud Shell](https://shell.azure.com/powershell) (PowerShell):
 
 `Invoke-WebRequest -Uri "https://raw.githubusercontent.com/azure-samples/distributed-az-edge-framework/main/deployment/deploy-az-demo-bootstrapper.ps1" -OutFile "./deploy-az-demo-bootstrapper.ps1" && ./deploy-az-demo-bootstrapper.ps1`
 
 Optionally, for deploying a developer environment with local application building and deployment please see [Setup a development environment on Azure with local application deployment](./deployment/deploy-dev.md).
 
+### Validate Deployment
+
+After successful deployment, the following resources will be created in your Azure subscription:
+- Three resource groups with each having their own AKS cluster, separate and peered virtual networks and Azure Arc connected.
```
[the remark this answers is not in the document]
Comment (katriendg) [reply README.md]: Attempted... wdyt?

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (katriendg) [reply .devcontainer/devcontainer.json:27]: Done

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (katriendg) [reply README.md]: As discussed offline, this is now changed to a more correct name as it's not only for dev setup cleanup. New name is `cleanup-az-resources.ps1`

**Review by suneetnangia — APPROVED** [review]
LGTM, Thanks 👍

**Pull request merged**